packages/core/src/runtime-configurations.ts

```diff
@@ -0,0 +1,137 @@
+import type { IRunOptions, TopLevelConfig } from './index';
+declare global {
+    interface Window {
+        engineEntryOptions?: (options: { urlParams: IRunOptions; envName: string }) => IRunOptions;
+    }
+}
+
+export type ConfigLoader = () => Promise<
+    {
+        default: TopLevelConfig;
+    }[]
+>;
+
+export type ConfigLoaders = Record<string, ConfigLoader>;
+
+export class RuntimeConfigurations {
+    fetchedConfigs: Record<string, Promise<TopLevelConfig>> = {};
+    constructor(private envName: string, private loaders: ConfigLoaders) {
+        // validate args since we use this class in the entry point template code
+        if (!envName) {
+            throw new Error('envName must be provided');
+        }
+    }
+    private isMainWebEntrypoint() {
+        if (!(globalThis instanceof Window)) {
+            return false;
+        }
+        const win = globalThis as unknown as Window;
```

Comment:
avoid the cast to Window.
just do `(globalThis as {parent?: unknown}).parent === globalThis`